tests/benchmarks/latency_measure/src/timing_sc.c

```diff
@@ -37,17 +37,28 @@ static void start_thread_entry(void *p1, void *p2, void *p3)
 	uint32_t  num_iterations = (uint32_t)(uintptr_t)p1;
 	timing_t  start;
 	timing_t  finish;
+	uint64_t min_cycles = UINT64_MAX;
+	int key;
 
 	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
-	start = timing_timestamp_get();
-	for (uint32_t i = 0; i < num_iterations; i++) {
-		timing_timestamp_get();
+	key = irq_lock();
+	k_sched_lock();
```

Comment:
I like the idea of running this a few times to determine the minimum overhead value. However, locking interrupts and the scheduler using irq_lock() and k_sched_lock() respectively are not allowed from userspace. I note that some of the CI tests failed, and I suspect that this is the cause.

Admittedly I'm not certain what the source of the variance in the timings is--my initial suspicions are that a timer interrupt could be firing and getting serviced. If that is indeed the situation, then we can try to further reduce the likelihood of such an interruption by aligning the calibration to a tick boundary beforehand. The other item that comes to mind, is that some boards can not support a minimum system tick frequency of 1 Hz (they use an overlay to change it to a different value). I mention this as I have observed in the past that using a value too small for the timer driver to use can result in some messed up timing samples.

Reply:
I've removed the locks, relying on repeated calculation of the overhead is good enough in my test

> aligning the calibration to a tick boundary beforehand.

How do we do that?

Reply:
In general, there are two ways to align to a tick boundary, or close enough to one.  The first (and simplest) is to issue a call to k_sleep(). When the thread wakes, assuming that no other thread is going to take precedence, the thread should be (nearly) aligned to a tick boundary.

```
k_sleep(K_TICKS(1));
```

The second is to a brute-force busy-wait approach. It too assumes that no other thread will take precedence in scheduling.

```
uint32_t current_tick = sys_clock_tick_get_32();

while (sys_clock_tick_get_32() == current_ticks) {
}
```